src/lidar_subscriber.cpp

```diff
@@ -0,0 +1,156 @@
+  //std::cout<<first.bearing_rad<<" " <<second.bearing_rad<<"\n";
+  return average;
+}
+
+class LidarSubscriber : public rclcpp::Node
+{
+  public:
+    LidarSubscriber()
+   : Node("lidar_subscriber")
+    {
+      subscription = this->create_subscription<sensor_msgs::msg::LaserScan>(
+      "scan_filtered", 10, std::bind(&LidarSubscriber::laserscan_to_landmarks, this, std::placeholders::_1));
+      publisher = this->create_publisher<visualization_msgs::msg::Marker>("landmark_visualization",10);
+    }
+  private:
+    rclcpp::Subscription<sensor_msgs::msg::LaserScan>::SharedPtr subscription;
+    rclcpp::Publisher<visualization_msgs::msg::Marker>::SharedPtr publisher;
+    std::queue<Observation2D> laserscan_to_landmarks(const sensor_msgs::msg::LaserScan::SharedPtr msg){
+        float angle_increment = msg->angle_increment;
+        int measurement_count = static_cast<int>(std::round((msg->angle_max-msg->angle_min)/angle_increment)) + 1;
+        std::queue<Observation2D> landmarks;
+        std::queue<int> counts; 
+	struct Observation2D previous_measurement;
+	struct Observation2D first_measurement;
+        for (int i = 1; i < measurement_count; i++){
+            float curr_range = msg->ranges[i];
+            if (curr_range == std::numeric_limits<float>::infinity()||curr_range == -1*std::numeric_limits<float>::infinity()){ 
+           continue;
+	    }else{
+                float curr_bearing = i * angle_increment;
+		struct Observation2D curr_measurement = {curr_range, curr_bearing};
+	       if (landmarks.empty()){
+		       landmarks.push(curr_measurement);
+		       previous_measurement = curr_measurement;
+		       first_measurement = curr_measurement;
+		       counts.push(1);	       
+	       }else if(same_landmark(curr_measurement, previous_measurement)){
+                  counts.back()++;
+                  struct Observation2D updated_landmark = merge(landmarks.back(), curr_measurement, counts.back());
+                  (landmarks.back()).range_m = updated_landmark.range_m;
+                  (landmarks.back()).bearing_rad = updated_landmark.bearing_rad; 
+                }else{
+                  landmarks.push(curr_measurement);
+                  counts.push(1);
+                }
+	       previous_measurement = curr_measurement;
+            }
+        }
+	struct Observation2D last_measurement = previous_measurement;
+        //if the first and last landmarks are actually the same, they get combined!
+	if (landmarks.size()>1 and same_landmark(first_measurement, last_measurement)){
+          struct Observation2D new_last = merge(landmarks.front(), landmarks.back(), counts.front(), counts.back());
+          (landmarks.back()).range_m = new_last.range_m;
+          (landmarks.back()).bearing_rad = new_last.bearing_rad; 
+          landmarks.pop();
+        }
+	
+std::cout<<landmarks.size()<<" "<<landmarks.front().range_m<<" "<<landmarks.front().bearing_rad<<"\n";
```

Comment:
ROS2 also has a built in logging framework. You should replace your `cout` print statement here with the ROS debug prints.